packages/core/src/prompts/prompt.ts

```diff
@@ -172,7 +177,17 @@ export default class Prompt {
 
 		if (key?.name === 'return') {
 			if (this.opts.validate) {
-				const problem = this.opts.validate(this.value);
+				this.state = 'validate';
```

Comment:
When the `race`/`timeout` is sorted, shouldn't this move closer to the `render`? No need to set it if never rendered?